src/bigstring_unix_stubs.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2019, Christopher Zimmermann
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
+ * SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
+ * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+ * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+
+#include <unistd.h>
+#include <caml/mlvalues.h>
+#include <caml/bigarray.h>
+#include <caml/unixsupport.h>
+#include <caml/threads.h>
+
+CAMLprim value
+bigstring_read(value vfd, value vba, value voff, value vlen)
+{
+    void *iobuf = ((char *)Caml_ba_data_val(vba)) + Unsigned_long_val(voff);
```

Comment:
There should be some `Caml_param` macro here, at least for `vfd` since it's a heap allocated object, right?

Reply:
I made sure not to dereference any pointer into the OCaml heap after ``caml_release_runtime_system()``. I don't do any allocations. So no need to register pointers into the OCaml heap with the Gc. That's the nice thing about bigarrays :)